### DISCUSSION TITLE
memory consumption [question]

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#437] [Q&A]
my moosefs cluster naturally grows with memory usage
no surprise here

to address it I would like to figure out what eats it.
is there a way to check memory consumption by folder? etc?

**Comment by chogata** [comment]
You can see memory consumption on your INFO in CGI, there is a whole table titled "Memory usage detailed info" that tells you how your memory is used. You ask about "memory consumption by folder" - I assume you mean folders on MooseFS? That is not possible and would not make much sense anyway - MooseFS stores filesystem metadata info in some memory structures, chunk data info in other structures and combining the two is not possible, when you have hardlinks or snapshots in your data.
The size of RAM corresponds directly to: number of filesystem objects (all inodes), number of chunks and is also affected by length of info about inodes (files and directory names).
Also, deleting some of the data will not currently free up your RAM, not without restarting the Master Server. Since memory allocation in various systems is not very efficient, MooseFS has its own internal memory manager and allocates big(ish) chunks of memory, which it then uses as needed. And it doesn't like to "give it back", so to speak ;) Of course, it reuses memory, so if you delete half your data and then record some new data, the "unused" memory will be reused first, before new memory will be allocated. Nevertheless, to free memory after deleting a lot of data you have to currently restart the master server. We have it on our roadmap to change this a bit, so that if a lot of memory is unused, MooseFS will free it up. But until then you need a restart to really free it.
